django/db/migrations/serializer.py

```diff
@@ -270,6 +272,40 @@ def serialize(self):
         return "uuid.%s" % repr(self.value), {"import uuid"}
 
 
+class ClassSerializer(BaseSerializer):
+    def serialize(self):
+        _module = self.value.__module__
+        module_first_part, module_second_part = _module.split('.')
+        _class = self.value.__class__
+        _path = ""
+        if module_second_part.startswith('_'):
+            related_modules = [
+                m for m in sys.modules
+                if module_first_part in m and
+                m != _module and len(m.split('.')) > 1 and not
+                m.split('.')[1].startswith('_')
+            ]
+            for related_module in related_modules:
+                klass_members = dict(
+                    inspect.getmembers(
+                        sys.modules[related_module],
+                        lambda m: inspect.isclass(m))).keys()
+                if _class.__name__ not in klass_members:
+                    continue
+                _path = related_module + "." + _class.__name__
+                break
+        from django.utils.deconstruct import deconstructible
+        deconstructible_klass = deconstructible(_class, path=_path)
+        try:
+            init_values = self.value.__reduce__()
+            new_value = deconstructible_klass()
+            if len(init_values) > 2:
+                new_value = deconstructible_klass(init_values[2])
+            return DeconstructableSerializer(new_value).serialize()
+        except TypeError:
+            pass  # not suitable for deconstructible
```

Comment:
This is the sole `BaseSerializer` subclass that returns `None` on `serialize` failure.